makefiles/features_precheck.inc.mk

```diff
@@ -0,0 +1,17 @@
+# Check if provided features are available
+
+ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)) $(filter newlib,$(FEATURES_PROVIDED)))
+  ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)))
```

Comment:
You need to white-list this file in `dist/tools/buildsystem_sanity_check/check.sh:75`. Please also update the comment in the line above.

Reply:
ack

Reply:
> You need to white-list this file in `dist/tools/buildsystem_sanity_check/check.sh:75`. Please also update the comment in the line above.

Huh? This completely bypasses the intention of the check...

Reply:
No, this was added when `FEAUTES_OPTIONAL` was added to replace the paddern

```Makefile
ifneq (,$(filter foo,$(FEAUTRES_PROVIDES)))
  USEMODULE += optional_submodule
endif
```

The intention was to enforce the use of `FEATURES_OPTIONAL` instead. But this here is unrelated.

Reply:
> No, this was added when `FEAUTES_OPTIONAL` was added to replace the paddern
> 
> ```makefile
> ifneq (,$(filter foo,$(FEAUTRES_PROVIDES)))
>   USEMODULE += optional_submodule
> endif
> ```
> 
> The intention was to enforce the use of `FEATURES_OPTIONAL` instead. But this here is unrelated.

I'm not sure I understand, `FEATURES_OPTIONAL` should not be checked either, only `FEATURES_USED`. The buildsystem check is there so that there are no conditionals against `FEATURES_OPTIONAL`, `FEATURES_PROVIDED`, `FETURES_REQUIRED`, see original or #10179 and https://github.com/RIOT-OS/RIOT/blob/3a9f8d5851b0990fd709950b65341dd2e8cb49f9/dist/tools/buildsystem_sanity_check/check.sh#L56-L75, am I looking at the wrong line?